app/lib/audit/catalog_to_archive.rb

```diff
@@ -0,0 +1,71 @@
+module Audit
+  # Catalog to cloud archive provider (what ZipEndpoint, ZippedMoabVersion etc represent) audit code
+  class CatalogToArchive
+    # delegate :bucket, :bucket_name, to: PreservationCatalog::S3
+
+    def self.logger
+      @logger ||= Logger.new(STDOUT)
+                        .extend(ActiveSupport::Logger.broadcast(Logger.new(Rails.root.join('log', 'c2a.log'))))
+    end
+
+    def self.check_aws_replicated_zipped_moab_version(zmv)
+      return unless check_child_zip_part_attributes(zmv)
+
+      zmv.zip_parts.where.not(status: :unreplicated).each do |part|
+        aws_s3_object = PreservationCatalog::S3.bucket.object(part.s3_key)
+        if aws_s3_object.exists?
+          part.update(last_existence_check: Time.zone.now)
+        else
+          bucket_name = PreservationCatalog::S3.bucket_name
+          logger.error("Archival Preserved Copy: #{zmv.inspect} #{part.inspect} was not found on #{bucket_name}.")
+          part.update(status: 'not_found', last_existence_check: Time.zone.now)
+          next
+        end
+
+        replicated_checksum = replicated_checksum(aws_s3_object)
+        if part.md5 == replicated_checksum
+          part.update(last_checksum_validation: Time.zone.now)
+        else
+          logger.error("Stored checksum(#{part.md5}) doesn't match the replicated checksum(#{replicated_checksum}).")
```

Comment:
maybe a comment making it explicit that no checksum computation is happening, this is just checking metadata that we gave AWS and that we also still store, for drift between the two systems.  we never expect that to differ, but we're here, and it's a cheap check, so why not?  because it'd be pretty weird if they did differ.